File: weather/src/main/java/com/nerdery/pkarels/weather/data/model/TempUnit.java
package com.nerdery.pkarels.weather.data.model;

import com.google.gson.annotations.SerializedName;

/**
 * Temperature unit to be used in requests for {@link com.nerdery.pkarels.weather.data.api.WeatherService}
 */
public enum TempUnit {
    @SerializedName("si")
    CELSIUS("si"),

    @SerializedName("us")
    FAHRENHEIT("us");

    private String value;

    TempUnit(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: weather/src/main/java/com/nerdery/pkarels/weather/model/TempUnit.kt
package com.nerdery.pkarels.weather.model

import com.google.gson.annotations.SerializedName

/**
 * Temperature unit to be used in requests for [com.nerdery.pkarels.weather.data.api.WeatherService]
 */
enum class TempUnit constructor(private val value: String) {
    @SerializedName("si")
    CELSIUS("si"),

    @SerializedName("us")
    FAHRENHEIT("us");

    override fun toString(): String {
        return value
    }
}
